lib/core/job_cleanup.rb

```diff
@@ -0,0 +1,65 @@
+#
+# Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+# or more contributor license agreements. Licensed under the Elastic License;
+# you may not use this file except in compliance with the Elastic License.
+#
+
+# frozen_string_literal: true
+
+require 'core'
+require 'utility/logger'
+
+module Core
+  class JobCleanUp
+    class << self
+      def execute(connector_id = nil)
+        process_orphaned_jobs
+        process_stuck_jobs(connector_id)
+      end
+
+      private
+
+      def process_orphaned_jobs
+        Utility::Logger.info('Start cleaning up orphaned jobs...')
+        orphaned_jobs = ConnectorJob.orphaned_jobs
+        if orphaned_jobs.empty?
+          Utility::Logger.info('No orphaned jobs found. Skipping...')
+          return
+        end
+
+        # delete content indicies in case they are re-created by sync job
+        content_indices = orphaned_jobs.map(&:index_name).compact.uniq
+        ElasticConnectorActions.delete_indices(content_indices) if content_indices.any?
+        result = ConnectorJob.delete_jobs(orphaned_jobs)
+        Utility::Logger.error("Error found when deleting jobs: #{result['failures']}") if result['failures']&.any?
+        Utility::Logger.info("Successfully deleted #{result['deleted']} out of #{result['total']} orphaned jobs.")
+      end
+
+      def process_stuck_jobs(connector_id = nil)
+        Utility::Logger.info("Start cleaning up stuck jobs for #{connector_id ? "connector #{connector_id}" : 'native connectors'}...")
```

Comment:
```suggestion
        Utility::Logger.debug("Start cleaning up stuck jobs for #{connector_id ? "connector #{connector_id}" : 'native connectors'}...")
```

Reply:
You are the 2nd one who suggests I change it to `debug`.

As orphaned/stuck jobs are rare, I think it makes sense. 

I will leave below as it is, so that we will know how many jobs are processed if any.

https://github.com/elastic/connectors-ruby/blob/64d612512fa214ea5758322705f5609435aaaa4f/lib/core/job_cleanup.rb#L34-L35

https://github.com/elastic/connectors-ruby/blob/64d612512fa214ea5758322705f5609435aaaa4f/lib/core/job_cleanup.rb#L61